pkg/asset/installconfig/azure/validation.go

```diff
@@ -54,25 +54,26 @@ func Validate(client API, ic *types.InstallConfig) error {
 }
 
 // ValidateInstanceType ensures the instance type has sufficient Vcpu and Memory.
-func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceType string, diskType string, req resourceRequirements) field.ErrorList {
+func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceType string, diskType string, hyperVVersion string, req resourceRequirements) (string, field.ErrorList) {
```

Comment:
It is a smell to have a validation function returning anything other than whether the inputs passed the validation. Perhaps we should be separate the functionality of gathering from the cloud provider the information for the instance type from the functionality of validating that information against the requirements.